test/e2e/apimachinery/openapiv3.go

```diff
@@ -138,12 +174,24 @@ var _ = SIGDescribe("OpenAPIV3", func() {
 		framework.ExpectNoError(err)
 		aggrclient, err := aggregatorclient.NewForConfig(config)
 		framework.ExpectNoError(err)
+		c := openapi3.NewRoot(f.ClientSet.Discovery().OpenAPIV3())
+		gv := schema.GroupVersion{Group: samplev1beta1.GroupName, Version: "v1beta1"}
 		names := generateSampleAPIServerObjectNames(f.Namespace.Name)
 		SetUpSampleAPIServer(ctx, f, aggrclient, imageutils.GetE2EImage(imageutils.APIServer), names, samplev1beta1.GroupName, "v1beta1")
-		defer cleanupSampleAPIServer(ctx, f.ClientSet, aggrclient, names, "v1beta1.wardle.example.com")
+		defer func() {
+			cleanupSampleAPIServer(ctx, f.ClientSet, aggrclient, names, "v1beta1.wardle.example.com")
+			// Poll for the OpenAPI to be updated with the deleted aggregated apiserver.
+			err = wait.PollUntilContextTimeout(ctx, time.Second*1, wait.ForeverTestTimeout, true, func(_ context.Context) (bool, error) {
```

Comment:
same as above. Waiting here probably means coupling of tests that we rather avoid in the first place.